guvnor-structure/guvnor-structure-backend/src/main/java/org/guvnor/structure/backend/repositories/PullRequestServiceImpl.java

```diff
@@ -0,0 +1,265 @@
+    }
+
+    @Override
+    public PullRequest rejectPullRequest( final String repository,
+                                          final long id ) {
+        this.changeStatusPullRequest( repository, id, PullRequestStatus.REJECTED );
+        return this.getRepositoryMetadata( repository ).getPullRequest( id );
+    }
+
+    @Override
+    public PullRequest closePullRequest( final String repository,
+                                         final long id ) {
+        this.changeStatusPullRequest( repository, id, PullRequestStatus.CLOSED );
+        return this.getRepositoryMetadata( repository ).getPullRequest( id );
+    }
+
+    @Override
+    public List<PullRequest> getAllPullRequestsFor( final String repository,
+                                                    final String branch ) {
+        GitMetadata metadata = getRepositoryMetadata( repository );
+        List<PullRequest> pullRequests = metadata.getPullRequests( elem -> elem.getTargetBranch().equals( branch ) );
+        return pullRequests;
+    }
+
+    @Override
+    public List<PullRequest> getAllPullRequestsFor( final String repository ) {
+
+        GitMetadata metadata = getRepositoryMetadata( repository );
+        List<PullRequest> pullRequests = metadata.getPullRequests();
+        return pullRequests;
+
+    }
+
+    @Override
+    public List<PullRequest> getAllPullRequestsFor( final String repository,
+                                                    final PullRequestStatus status ) {
+
+        final List<PullRequest> pullRequests = this.getAllPullRequestsFor( repository );
+        return pullRequests.stream().filter( elem -> elem.getStatus().equals( status ) ).collect( Collectors.toList() );
+
+    }
+
+    @Override
+    public void deletePullRequest( final String repository,
+                                   final long id ) {
+
+        checkNotEmpty( "repository", repository );
+
+        GitMetadata metadata = getRepositoryMetadata( repository );
+        PullRequest pullRequest = metadata.getPullRequest( id );
+
+        List<PullRequest> finalPullRequests = metadata.getPullRequests();
+        finalPullRequests.remove( pullRequest );
+
+        GitMetadata storableMetadata = new GitMetadataImpl( metadata.getName(), metadata.getOrigin(), metadata.getForks(), finalPullRequests );
+        this.metadataStore.write( storableMetadata.getName(), storableMetadata );
+
+        this.deleteHiddenBranch( pullRequest );
+
+    }
+
+    @Override
+    public List<FileDiff> diff( final PullRequest pullRequest ) {
+
+        final Repository repository = configuredRepositories.getRepositoryByRepositoryAlias( pullRequest.getTargetRepository() );
+        String diff = String.format( "diff:%s,%s", pullRequest.getTargetBranch(), pullRequest.getSourceBranch() );
+        return (List<FileDiff>) this.ioService.readAttributes( convert( repository.getRoot() ), diff );
+    }
+
+    public void changeStatusPullRequest( final String repository,
```

Comment:
If this is ```public``` should it declared in the interface and annotated ```@Override```. If you don't really want it public (e.g. for tests) perhaps package-protected is a better modifier?